main.cpp

```diff
@@ -92,6 +92,129 @@ good to go!
  */
 
 #include <iostream>
+
+ /*4) make them return the correct primitive type. e.g. if you're implementing the FloatType::add function, your implementation would start like this:
+        float FloatType::add( float lhs, float rhs ) */
+
+#include <math.h>
+#ifdef INFINITY
+#endif
+
+
+///////// FLOAT  ...............
+///////// 
+class FloatType 
+{
+    public:
+        float lhs, rhs;
+        float add(float, float);
+        float subtract(float, float);
+        float multiply(float, float);
+        float divide(float, float);
+};
+
+float FloatType::add(float a, float b)
+{
+    return a + b;
+}
+
+float FloatType::subtract(float a, float b)
+{
+    return a - b;
+}
+
+float FloatType::multiply(float a, float b)
+{
+    return a * b;
+}
+
+float FloatType::divide(float a, float b)
+{
+    if (b<1e-7f && b>-1e-7f)
+        {
+        std::cout <<std::endl<<"warning, floating point division by zero returns 'inf' !" <<std::endl;
+            return(a/b);
+        }
+    return a/b;
+}
+
+
+
+///////// DOUBLE ...............
+///////// 
+class DoubleType 
+{
+    public:
+        double add(double, double);
+        double subtract(double, double);
+        double multiply(double, double);
+        double divide(double, double);
+};
```

Comment:
use `struct` not `class`
Accessor keywords have not been explained at this point in the course.
`struct` is `public` by default
Fix the indentation when you remove the word `public` too.
- [ ] revise